### PR TITLE
Extract functionality for creating and manipulating a project should be moved from the app to the lib

### PR DESCRIPTION
**Pull request by andped10**
New `Project` class to hold the models, experiments and the other elements that make up a full analysis.

`unique_name` need to be passed to constructor to recreate object

New way to create the default collection for the various classes.

### CODE REVIEW

File: src/easyreflectometry/project.py [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+from easyreflectometry.sample import Layer
+from easyreflectometry.sample import MaterialCollection
+from easyreflectometry.sample import Multilayer
+from easyreflectometry.sample import Sample
+from easyreflectometry.sample.collections.base_collection import BaseCollection
+
+
+class Project:
+    def __init__(self):
+        self._info = self._defalt_info()
```
Comment (rozyczko) [inline src/easyreflectometry/project.py:24]: typo -> `_default_info`

**Review by rozyczko — APPROVED** [review]
So this looks very close to the Job implementation in the diffraction library.
The `DiffractionJob` class doesn't (yet) have any serialization functionality, as your `Project` does but it will be added.
These probably should be a single class with merged functionality.

File: pyproject.toml [hunk 2]
```diff
@@ -29,7 +29,7 @@ classifiers = [
 ]
 requires-python = ">=3.9,<3.13"
 dependencies = [
-    'easyscience @ git+https://github.com/EasyScience/EasyScience.git@develop',
+    'easyscience @ git+https://github.com/EasyScience/EasyScience.git@unique_name_bug',
```
Comment (henrikjacobsenfys) [inline pyproject.toml:32]: Can be changed back to develop once the pull request has been merged :) 
putting this only as a reminder

File: tests/sample/collections/test_base_collection.py [hunk 3]
```diff
@@ -63,19 +63,3 @@ def test_as_dict(self):
         assert p.as_dict()['name'] == 'name'
         assert len(p.as_dict()['data']) == 1
         assert p.as_dict()['data'][0]['name'] == 'layer'
-
-    def test_make_default_collection(self):
```
Comment (henrikjacobsenfys) [inline tests/sample/collections/test_base_collection.py:67]: Should the new DEFAULT_ELEMENTS be tested instead?

File: src/easyreflectometry/project.py (hunk 1 above)
Comment (andped10) [reply src/easyreflectometry/project.py]: fixed

File: tests/sample/collections/test_base_collection.py (hunk 3 above)
Comment (andped10) [reply tests/sample/collections/test_base_collection.py:67]: They are indirectly tested in the `test_default` for each of the collections.

I have added tests to confirm that we are also able to create empty collections.

File: pyproject.toml (hunk 2 above)
Comment (andped10) [reply pyproject.toml]: done

**Pull request merged**